src/bulletin/bulletin.service.ts

```diff
@@ -69,4 +44,79 @@ export class BulletinService {
       }
     });
   }
+
+  async updateBulletin(
+    id: number,
+    updateBulletinDto: BulletinDTO,
+    pdfAttachments: Express.Multer.File[],
+  ) {
+    return this.prismaService.$transaction(async (tx) => {
+      try {
+        const { deleted_attachment_ids, ...bulletinData } = updateBulletinDto;
+        const updatedBulletin = await tx.bulletin.update({
+          where: { id },
+          data: {
+            ...bulletinData,
+          },
+        });
+        const attachments = [];
+        // Check if is there any deleted attachment ids to delete
+        // Reducing the query time by doing promise.all to simultaneously update the given ids
+        if (!isEmpty(deleted_attachment_ids)) {
+          const existingAttachments = await tx.pDFAttachment.findMany({
+            where: {
+              id: { in: deleted_attachment_ids },
+              deleted_at: null,
+              bulletin_id: id,
+            },
+          });
+          const existingAttachmentIdsSet = new Set(
+            existingAttachments.map((att) => att.id),
+          );
+          const nonExistingIds = deleted_attachment_ids.filter(
+            (id) => !existingAttachmentIdsSet.has(id),
+          );
+          if (!isEmpty(nonExistingIds)) {
+            throw new Error(
+              `Attachments with IDs ${nonExistingIds.join(
+                ', ',
+              )} do not exists to bulletin ID: ${id}`,
+            );
+          }
+
+          await Promise.all(
+            deleted_attachment_ids.map((deleteID) =>
+              tx.pDFAttachment.update({
+                where: { id: deleteID },
+                data: { deleted_at: new Date() },
+              }),
+            ),
+          );
+        }
+        // Check if there is new pdf attachments
+        // Upload all and create new data for pdfAttacment table
+        if (!isEmpty(pdfAttachments)) {
```

Comment:
same here, Instead of creating PDF attachments one by one, you can use `createMany`

Reply:
nah impossible to do that, I process both uploading to storage and creating the data sa pdfattachment.

I need to confirm na ma upload ang pdf sa db then I create pdfattachment data.

I can separate the upload data sa pdf then create many. but thats another process naput